libevm/VMCalls.cpp

```diff
@@ -139,12 +142,20 @@ void VM::caseCreate()
 void VM::caseCall()
 {
 	m_bounce = &VM::interpretCases;
+
+	// TODO: Please check if that does not actually increases the stack size.
+	//       That was the case before.
 	unique_ptr<CallParameters> callParams(new CallParameters());
+
+	// Clear the return data buffer. This will not free the memory.
+	m_returnData.clear();
+
 	bytesRef output;
 	if (caseCallSetup(callParams.get(), output))
 	{
 		std::pair<bool, owning_bytes_ref> callResult = m_ext->call(*callParams);
 		callResult.second.copyTo(output);
+		m_returnData.assign(callResult.second.begin(), callResult.second.end());
```

Comment:
I'm not a big fun of helpers like `toBytes`, usually C++ std lib should be enough. But the question is: should we keep the full memory buffer (owning_bytes_ref) or copy only the return buffer.